File: src/main/java/ru/ecospas/domain/model/Organization.java
package ru.ecospas.domain.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@ToString(exclude = "user")
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Entity
@Table(name = "organization")
public class Organization implements BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @EqualsAndHashCode.Include
    private Integer id;

    @Column(name = "organization_full_name")
    private String organizationName;

    private String organizationShortName;
    private String organizationTypeActivity;
    private boolean oneTerritory;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @OneToMany(mappedBy = "organization")
    private List<ObjectModel> objects = new ArrayList<>();

    @OneToOne(
            mappedBy = "organization",
            cascade = CascadeType.ALL,
            orphanRemoval = true
    )
    private OrganizationAddress address;

    @OneToMany(
            mappedBy = "organization",
            cascade = CascadeType.ALL,
            orphanRemoval = true)
    private List<OrganizationSigner> signers = new ArrayList<>();

    @OneToMany(
            mappedBy = "organization",
            cascade = CascadeType.ALL,
            orphanRemoval = true)
    private List<OrganizationContact> contacts = new ArrayList<>();
}

File: frontend/src/types/organization.ts
export interface OrganizationFull {
    id?: number
    organizationName: string
    organizationShortName: string
    organizationTypeActivity: string
    oneTerritory: boolean
    address: {
        addressIndex: string
        constituentEntity: string
        city: string
        street: string
        house: string
    } | null
    signers: Array<{
        id?: number
        name: string
        position: string
        isPrimary?: boolean
    }>
    contacts: Array<{
        id?: number
        fullName: string
        position: string
        phones: string
        address: string
    }>
}